GetStream/stream-chat-android
# Add a way to convert Flow into LiveData for Java users

closes #3564 

### 🎯 Goal

Add a way to convert `Flow` into `LiveData` for Java users.

### 🛠 Implementation details
Added `Flow<T>.asLiveData` extension that uses `import androidx.lifecycle.asLiveData` under the hood.
The extension function is only available from the Java codebase.

### 🧪 Testing

1. Apply the patch below (it adds observing connection state from Java code)
2. Open the app and sign in as a user
3. Filter logcat for `RafalTest` - check logs are there

<details>

<summary>Provide the patch summary here</summary>

```
Index: stream-chat-android-ui-components-sample/src/main/kotlin/io/getstream/chat/ui/sample/application/ChatInitializer.kt
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/stream-chat-android-ui-components-sample/src/main/kotlin/io/getstream/chat/ui/sample/application/ChatInitializer.kt b/stream-chat-android-ui-components-sample/src/main/kotlin/io/getstream/chat/ui/sample/application/ChatInitializer.kt
--- a/stream-chat-android-ui-components-sample/src/main/kotlin/io/getstream/chat/ui/sample/application/ChatInitializer.kt	(revision f001bd775dfc76d6592393f28240721274712816)
+++ b/stream-chat-android-ui-components-sample/src/main/kotlin/io/getstream/chat/ui/sample/application/ChatInitializer.kt	(date 1657696389410)
@@ -31,6 +31,7 @@
 import io.getstream.chat.android.pushprovider.xiaomi.XiaomiPushDeviceGenerator
 import io.getstream.chat.android.ui.ChatUI
 import io.getstream.chat.ui.sample.BuildConfig
+import io.getstream.chat.ui.sample.TestClass
 import io.getstream.chat.ui.sample.feature.HostActivity
 
 @OptIn(InternalStreamChatApi::class)
@@ -75,5 +76,7 @@
 
         // Using markdown as text transformer
         ChatUI.messageTextTransformer = MarkdownTextTransformer(context)
+
+        TestClass().observeTest(client)
     }
 }
Index: stream-chat-android-ui-components-sample/src/main/java/io/getstream/chat/ui/sample/TestClass.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/stream-chat-android-ui-components-sample/src/main/java/io/getstream/chat/ui/sample/TestClass.java b/stream-chat-android-ui-components-sample/src/main/java/io/getstream/chat/ui/sample/TestClass.java
new file mode 100644
--- /dev/null	(date 1657696402856)
+++ b/stream-chat-android-ui-components-sample/src/main/java/io/getstream/chat/ui/sample/TestClass.java	(date 1657696402856)
@@ -0,0 +1,18 @@
+package io.getstream.chat.ui.sample;
+
+import io.getstream.chat.android.client.ChatClient;
+import io.getstream.chat.android.client.extensions.FlowExtensions;
+import io.getstream.chat.android.client.logger.ChatLogger;
+import io.getstream.chat.android.offline.extensions.ChatClientExtensions;
+import io.getstream.chat.android.offline.model.connection.ConnectionState;
+import kotlinx.coroutines.flow.StateFlow;
+
+public class TestClass {
+
+    public void observeTest(ChatClient chatClient) {
+        StateFlow<ConnectionState> connectionDataStateFlow = ChatClientExtensions.getGlobalState(chatClient).getConnectionState();
+        FlowExtensions.asLiveData(connectionDataStateFlow).observeForever(connectionState -> {
+            ChatLogger.Companion.get("RafalTest").logE("Test " + connectionState);
+        });
+    }
+}

```

</details>


### ☑️Contributor Checklist

#### General

#### Code & documentation

### ☑️Reviewer Checklist
